adbms/src/test/scala/de/up/hpi/informationsystems/adbms/definition/FutureRelationTest.scala

```diff
@@ -0,0 +1,270 @@
+                .withCellContent(colOrderId)(505)
+                .withCellContent(colOrderdate)("08/06/07")
+                .build(),
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colCustomerId)(7)
+                .withCellContent(colFullname)("Omari Wesson")
+                .withCellContent(colCountry)("USA")
+                .build()
+            ))
+        }
+
+        "return the appropriate result set for an innerJoin" in {
+          orders
+            .innerJoin(customers, (left, right) => left.get(colCustomerId) == right.get(colCustomerId))
+            .records shouldEqual
+            Success(Seq(
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colCustomerId)(14)
+                .withCellContent(colFullname)("Max Mustermann")
+                .withCellContent(colCountry)("Germany")
+                .withCellContent(colOrderId)(504)
+                .withCellContent(colOrderdate)("05/06/07")
+                .build(),
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colCustomerId)(14)
+                .withCellContent(colFullname)("Max Mustermann")
+                .withCellContent(colCountry)("Germany")
+                .withCellContent(colOrderId)(505)
+                .withCellContent(colOrderdate)("08/06/07")
+                .build()
+            ))
+        }
+
+        "return the appropriate result set for an outerJoin" in {
+          orders
+            .outerJoin(customers, (left, right) => left.get(colCustomerId) == right.get(colCustomerId))
+            .records shouldEqual
+            Success(Seq(
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colCustomerId)(14)
+                .withCellContent(colFullname)("Max Mustermann")
+                .withCellContent(colCountry)("Germany")
+                .withCellContent(colOrderId)(504)
+                .withCellContent(colOrderdate)("05/06/07")
+                .build(),
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colCustomerId)(14)
+                .withCellContent(colFullname)("Max Mustermann")
+                .withCellContent(colCountry)("Germany")
+                .withCellContent(colOrderId)(505)
+                .withCellContent(colOrderdate)("08/06/07")
+                .build(),
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colOrderId)(504)
+                .withCellContent(colOrderdate)("17/06/07")
+                .withCellContent(colCustomerId)(6)
+                .build(),
+              Record(orderColumns ++ customerColumns)
+                .withCellContent(colCustomerId)(7)
+                .withCellContent(colFullname)("Omari Wesson")
+                .withCellContent(colCountry)("USA")
+                .build()
+            ))
+        }
+      }
+    }
+  }
+}
```

Comment:
I personally think, we should move all of those to `TransientRelationTest` as this logic is implemented there. This test can then test the parallel execution of those operations and additional operators that will follow.

Reply:
But we already have these tests for `TransientRelation` in there. How would you tests parallel execution of them there?

Reply:
Don't know yet :laughing: